### PR TITLE
SEC-2557 Enabled configuration of GrantedAuthoritiesMapper in LdapAuthenticationProviderConfigurer

### PR DESCRIPTION
**Pull request by lukashinsch**
Since the original PR #82 for https://jira.spring.io/browse/SEC-2557 needed some git cleanup I took the liberty to extract the relevant original commit by @dalbrekt and post as a new PR.

(I did not change the code but signed the contributors agreement just in case)

### CODE REVIEW

**Comment by dalbrekt** [comment]
:+1:

File: config/src/main/java/org/springframework/security/config/annotation/authentication/configurers/ldap/LdapAuthenticationProviderConfigurer.java [hunk 1]
```diff
@@ -134,6 +133,38 @@ private LdapAuthoritiesPopulator getLdapAuthoritiesPopulator() {
...
+		return this;
+	}
+
+	/**
+	 * Gets the {@link GrantedAuthoritiesMapper} and defaults to {@link SimpleAuthorityMapper}.
+	 *
+	 * @return the {@link GrantedAuthoritiesMapper}
+	 * @throws Exception if errors in {@link SimpleAuthorityMapper#afterPropertiesSet()}
+	 */
+	protected GrantedAuthoritiesMapper getAuthoritiesMapper() throws Exception {
```
Comment (rwinch) [inline config/src/main/java/org/springframework/security/config/annotation/authentication/configurers/ldap/LdapAuthenticationProviderConfigurer.java]: Can we change this to private please?

Comment (lukashinsch) [inline config/src/main/java/org/springframework/security/config/annotation/authentication/configurers/ldap/LdapAuthenticationProviderConfigurer.java]: The method was made accessible for testing. Should we remove the test case then? While I think it's commendable to have unit tests this one was probably not exactly the most important one to have (also being the only one for the builder class).

Comment (rwinch) [inline config/src/main/java/org/springframework/security/config/annotation/authentication/configurers/ldap/LdapAuthenticationProviderConfigurer.java]: We should be able to achieve testing by validating the LdapAuthenticationProvider

**Comment by pivotal-issuemaster** [comment]
@lukashinsch Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring)!

**Comment by pivotal-issuemaster** [comment]
@lukashinsch Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal)!

[Click here](null) to manually synchronize the status of this Pull Request.

See the [FAQ](null) for frequently asked questions.

**Comment by rwinch** [comment]
Pushing back since the PR has not been updated with the feedback requested

**Comment by eleftherias** [comment]
Thank you for the PR @lukashinsch. However, the same code changes have already been applied via b76e3be . Therefore, I am closing this PR.

**Pull request closed without merging**